src/lib/core/trigger.h

```diff
@@ -57,6 +60,19 @@ struct trigger
 	 * or the object containing the trigger is destroyed.
 	 */
 	trigger_f0 destroy;
+	/**
+	 * Flag used if triggers runs by trigger_fiber_run function,
+	 * otherwise not used.
+	 * In trigger_fiber_run function if this flag is set,
+	 * trigger runs in fiber asynchronously.
+	 */
+	bool async;
+	/**
+	 * Fiber in which the trigger is triggered,
+	 * used in the trigger_finger_run for
+	 * trigger_fiber_run_reverse functions
+	 */
+	struct fiber *fiber;
```

Comment:
I would be glad if we didn't change struct trigger.
This member is only used (set and cleared) only inside one function `trigger_fiber_run`. That means that we store pointer for all triggers for their whole lifetime while using only for on_shutdown triggers and only in shot perion of their life. I think it's overkill. I think we should allocate this pointer inside `trigger_fiber_run` (using region alloc).